tests/license_header_check.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+import sys
+from pathlib import Path
+
+paths = ["rust/src", "tests"]
+
+header_pattern = r"(?:\/\/|\#)\s*SPDX-License-Identifier: (LGPL-2\.1-or-later|Apache-2\.0)(?:\r?\n)?"
+
+excludes = [
+    "**/*.json",
+    "**/*.txt",
+    "**/*.in",
+    ".gitignore",
+    ".gitmodules",
+    "LICENSE",
+    "NOTICE",
+    "go.mod",
+    "go.sum",
+    ".git/*",
+    "logo/*",
+    "licenses",
+    "**/*.md",
+    "**/*.lock",
+    "**/*.toml",
+    "**/*.yml",
+    "tests/integration/test_802.1x_srv_files/*",
+    "tests/integration/test_ipsec_certs/*",
+    "tests/integration/test_captures/*",
+    "rust/target/*",
+    "rust/src/python/dist/*",
+    "rust/src/python/build/*",
+    "rust/src/python/nmstate.egg-info/*",
+    "examples/*",
+    "packaging/*",
+]
+
+total_files = 0
+files_with_header = 0
+files_without_header = 0
+excluded_files = 0
+
+
+def get_included_files():
+    global total_files
+    global excluded_files
```

Comment:
Nit, maybe instead of using the global variables, we should return values from functions and pass them as parameters where needed. This will make functions more self-contained and easier to understand. But it more sounds like personal taste.

Reply:
or should the test be wrapped with class